cmd/csi-proxy/main.go

```diff
@@ -68,7 +72,22 @@ func main() {
 	if err != nil {
 		panic(err)
 	}
-	s := server.NewServer(apiGroups...)
+
+	reg := prometheus.NewRegistry()
+	go func() error {
+		m := http.NewServeMux()
+		// Create HTTP handler for Prometheus metrics.
+		m.Handle("/metrics", promhttp.HandlerFor(
```

Comment:
It seems that there's a generic impl in the SMB CSI Driver 

https://github.com/kubernetes-csi/csi-driver-smb/blob/a71b8cf5b868091fdaa6f52842a1e8ae9fe6c98e/cmd/smbplugin/main.go#L122

Could we use https://github.com/kubernetes/component-base/tree/master/metrics too?